server/v1/controllers/shows/utils/utils.js

```diff
@@ -36,7 +36,28 @@ async function showList(shows, startDate = null, endDate = null) {
 
   //console.log(instanceKeyBy);
   //Combined series and instance shows by object ID and then return the final array
-  return { ...seriesKeyBy, ...instanceKeyBy };
+  let showsToReturn = { ...seriesKeyBy, ...instanceKeyBy };
+
+  //transform the object back to an array
+  let showsToReturnArray = [];
+  _.mapKeys(showsToReturn, function(value) {
+    showsToReturnArray.push(value);
+  });
+
+  //sort the array by event start time
+  showsToReturnArray = showsToReturnArray.sort(function(a, b) {
+    if (new Date(a.show_start_time_utc) > new Date(b.show_start_time_utc)) {
+      return 1;
+    } else if (
+      new Date(a.show_start_time_utc) === new Date(b.show_start_time_utc)
+    ) {
+      return 0;
+    } else {
+      return -1;
+    }
+  });
+
+  return showsToReturnArray;
```

Comment:
@jberns Are you okay with changing this to return an array instead of object? I think an array makes more sense based on what a user of the API would expect, and the object couldn't be sorted.

This code is adding ordering to the Find Shows API. The show calendar still works even though this is sending back an array - though I haven't tested it thoroughly so let me know if there's a specific thing I should look at. With these changes there is no master_time_id coming back from this API, so if that's being used somewhere, just let me know where and I'll get it fixed by adding master_time_id as a property to the objects being returned.

Reply:
I'm fine with returning this as an array.  It may have to be re-keyed on the front-end but that is easy to do in redux.

Also isn't the master_time_id is still one of the object properties?  It is not like we are deleting the property on the object when we key them.

I think everything works fine from the initial display, but the master_time_id is going to be used for making instances, updating shows, and deleting any type of show without requesting the entire time, instead everything is replaced in redux by the key based off of the single show return.

Reply:
@jberns ah you're totally right, `master_time_id` is still coming back! I'll leave this returning as an array.